processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryQueryToolChest.java

```diff
@@ -404,4 +405,39 @@ public static SegmentAnalysis finalizeAnalysis(SegmentAnalysis analysis)
         analysis.isRollup()
     );
   }
+
+  public EnumSet<SegmentMetadataQuery.AnalysisType> getAnalysisTypes(SegmentMetadataQuery query)
+  {
+    if (query.getAnalysisTypes() == null) {
+      return config != null ? config.getDefaultAnalysisTypes() : SegmentMetadataQueryConfig.DEFAULT_ANALYSIS_TYPES;
```

Comment:
It seems like handling null config shouldn't be necessary here. It shouldn't be null in production. If it is null sometimes in testing, how about replacing those spots with `new SegmentMetadataQueryConfig()` to get a default config? Then, `SegmentMetadataQueryConfig.DEFAULT_ANALYSIS_TYPES` could be private.